tests/std/tests/GH_001858_iostream_exception/test.cpp

```diff
@@ -148,6 +149,254 @@ void test_istream_exceptions() {
     }
 }
 
+template <typename CharT>
+CharT meow_array;
+template <>
+constexpr array<char, 5> meow_array<char> = {"meow"};
+template <>
+constexpr array<wchar_t, 5> meow_array<wchar_t> = {L"meow"};
+
+// testing GH-5070: basic_istream::get[line](char_type* s, std::streamsize n, char_type delim)
+// do not null-terminate the output buffer correctly
+template <class CharT>
+void test_gh5070_istream_get_nulltermination_under_exceptions() {
+    throwing_buffer<CharT> buffer;
+    const basic_string<CharT> stream_content(1U, meow_array<CharT>[2]);
+
+    { // get, exception during input extraction

```

Comment:
The `get` and `getline` comments are highly systematic, which is great, except that there's unnecessary variation in the first two, where `getline` is a bit more detailed:
```cpp
// getline, exception during input extraction, no exception rethrow
// getline, exception during input extraction, exception rethrow enabled
```